modules/recast/SCsub

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
+Import('env')
+
+# Not building in a separate env as core needs it
+
+# Thirdparty source files
+if (env['builtin_recast'] != 'no'):
+    thirdparty_dir = "#thirdparty/recastnavigation/Recast/"
```

Comment:
The Recast folder is included because the recastnavigation library includes a few sub-libraries - Recast, Detour, and DetourCrowd, mainly. This pull request only includes Recast but it's probably best to keep this architecture in case the others will be implemented later. 

You can see the third-party source here: https://github.com/recastnavigation/recastnavigation